core/src/main/java/org/infinispan/statetransfer/StateConsumerImpl.java

```diff
@@ -689,10 +686,13 @@ private CompletionStage<Void> applyChunk(Address sender, IntSet mySegments, Stat
       return CompletableFutures.completedNull();
    }
 
-   protected void onCompletedSegment(int segmentId) {
-      commitManager.stopTrackFor(PUT_FOR_STATE_TRANSFER, segmentId);
+   private void onCompletedSegment(int segmentId, InboundTransferTask inboundTransfer) {
```

Comment:
Should we use `Integer` here because of boxing in the map key?

Reply:
I don't think it has such a big impact to be honest. If we boxed it here, `CommitManager` has to unbox it.